subsys/bluetooth/controller/ll_sw/ull_peripheral_iso.c

```diff
@@ -35,3 +80,200 @@ int ull_peripheral_iso_reset(void)
 {
 	return 0;
 }
+
+uint8_t ull_peripheral_iso_acquire(struct ll_conn *acl,
+				   struct pdu_data_llctrl_cis_req *req,
+				   uint16_t *cis_handle)
+{
+	struct ll_conn_iso_group *cig;
+	struct ll_conn_iso_stream *cis;
+
+	/* Get CIG by id */
+	cig = ll_conn_iso_group_get_by_id(req->cig_id);
+	if (!cig) {
+		/* CIG does not exist - create it */
+		cig = ll_conn_iso_group_acquire();
+
+		memset(&cig->lll, 0, sizeof(cig->lll));
+
+		cig->cig_id = req->cig_id;
+		cig->lll.handle = 0xFFFF;
+		cig->lll.role = acl->lll.role;
+
+		for (int i = 0; i < CONFIG_BT_CTLR_CONN_ISO_STREAMS_PER_GROUP;
+		     i++) {
+			cig->cis_handles[i] = 0xFFFF;
+			cig->lll.cis_handles[cig->lll.num_cis] = 0xFFFF;
+		}
+
+		ull_hdr_init(&cig->ull);
+		lll_hdr_init(&cig->lll, cig);
+	}
+
+	if (cig->lll.num_cis == CONFIG_BT_CTLR_CONN_ISO_STREAMS_PER_GROUP) {
+		/* No space in CIG for new CIS */
+		return BT_HCI_ERR_INSUFFICIENT_RESOURCES;
+	}
+
+	/* Acquire new CIS */
+	cis = ll_conn_iso_stream_acquire();
+	if (cis == NULL) {
+		/* No space for new CIS */
+		return BT_HCI_ERR_INSUFFICIENT_RESOURCES;
+	}
+
+	cig->iso_interval = sys_le16_to_cpu(req->iso_interval);
+
+	cis->cis_id = req->cis_id;
+	cis->established = 0;
+	cis->group = cig;
+
+	cis->lll.acl_handle = acl->lll.handle;
+	cis->lll.sub_interval = sys_get_le24(req->sub_interval);
+	cis->lll.num_subevents = req->nse;
+	cis->lll.next_subevent = 0;
+	cis->lll.sn = 0;
+	cis->lll.nesn = 0;
+	cis->lll.cie = 0;
+
+	cis->lll.rx.phy = req->c_phy;
+	cis->lll.rx.burst_number = req->c_bn;
+	cis->lll.rx.flush_timeout = req->c_ft;
+	cis->lll.rx.max_octets = sys_le16_to_cpu(req->c_max_pdu);
+
+	cis->lll.tx.phy = req->p_phy;
+	cis->lll.tx.burst_number = req->p_bn;
+	cis->lll.tx.flush_timeout = req->p_ft;
+	cis->lll.tx.max_octets = sys_le16_to_cpu(req->p_max_pdu);
+
+
+	*cis_handle = ll_conn_iso_stream_handle_get(cis);
+	cig->lll.cis_handles[cig->lll.num_cis] = *cis_handle;
+	cig->cis_handles[cig->lll.num_cis] = *cis_handle;
+	cig->lll.num_cis++;
+
+	return 0;
+}
+
+uint8_t ull_peripheral_iso_setup(struct pdu_data_llctrl_cis_ind *ind,
+				 uint8_t cig_id, uint16_t cis_handle)
+{
+	struct ll_conn_iso_stream *cis = NULL;
+	struct ll_conn_iso_group *cig;
+
+	/* Get CIG by id */
+	cig = ll_conn_iso_group_get_by_id(cig_id);
+	if (!cig) {
+		return BT_HCI_ERR_UNSPECIFIED;
+	}
+
+	cig->lll.handle = ll_conn_iso_group_handle_get(cig);
+	cig->sync_delay = sys_get_le24(ind->cig_sync_delay);
+
+	/* Find CIS in CIG. NOTE: We could look up via handle, but we want to
+	 * sanity check the CIG/CIS link.
+	 */
+	for (int i = 0; i < CONFIG_BT_CTLR_CONN_ISO_STREAMS_PER_GROUP; i++) {
+		if (cig->cis_handles[i] == cis_handle) {
+			cis = ll_conn_iso_stream_get(cig->cis_handles[i]);
+			break;
+		}
+	}
+
+	if (!cis) {
+		return BT_HCI_ERR_UNSPECIFIED;
+	}
+
+	cis->sync_delay = sys_get_le24(ind->cis_sync_delay);
+	cis->lll.handle = cis_handle;
+	cis->lll.offset = sys_get_le24(ind->cis_offset);
+	cis->lll.event_count = 0;
+	cis->lll.event_count--;
```

Comment:
Why not just `cis->lll.event_count = -1`?

Reply:
I couldn't get the compiler to accept it, even by double cast. Did mean to fix it before upstreaming, but forgot about it. Problems most likely related to large bitfield variable - suggestions are welcome.

Reply:
I can't reproduce the problem locally now - maybe it was the CI system? Will try to change to "= -1" and submit